src/actions/arcActions.ts

```diff
@@ -8,16 +8,32 @@ import * as Redux from "redux";
 import { ThunkAction } from "redux-thunk";
 import * as Web3 from "web3";
 
-import * as notificationsActions from "actions/notificationsActions";
-import * as web3Actions from "actions/web3Actions";
+import { showAlert } from "actions/notificationsActions";
 import * as arcConstants from "constants/arcConstants";
-import * as web3Constants from "constants/web3Constants";
 import Util from "lib/util";
-import { IRootState } from "reducers";
+import { IRootState } from "reducers/index";
+import { IAccountState,
+         IDaoState,
+         IProposalState,
+         IStakeState,
+         ProposalStates,
+         TransactionStates,
+         VoteOptions } from "reducers/arcReducer";
+
 import * as schemas from "../schemas";
 
-import { IAccountState, IDaoState, IProposalState, IStakeState, ProposalStates, TransactionStates, VoteOptions } from "reducers/arcReducer";
-import { showAlert } from "actions/notificationsActions";
+export function loadCachedState() {
+  return async (dispatch: Redux.Dispatch<any>, getState: Function) => {
+    dispatch({ type: arcConstants.ARC_LOAD_CACHED_STATE_PENDING, payload: null });
+    try {
+      const cachedState = await axios.get('https://s3-us-west-2.amazonaws.com/daostack-alchemy/initialArcState-' + Arc.Config.get('network') + '.json');
+      dispatch({ type: arcConstants.ARC_LOAD_CACHED_STATE_FULFILLED, payload: cachedState.data });
```

Comment:
I'm having trouble figuring out where cachedState.data is being used to initialize the application data.  Can you please point me to it?

Reply:
in the reducer https://github.com/daostack/alchemy/pull/42/files#diff-10f59e3a1f2937ef1befd8f188e5345d

Reply:
all I see is `return payload`, but don't see what happens with it

Reply:
thats how Redux reducers work, that replaces that part of the state :)

Reply:
But where is the payload used?

Reply:
@dkent600 [here](https://github.com/daostack/alchemy/pull/42/files#diff-10f59e3a1f2937ef1befd8f188e5345dR124) where the reducer just returns the payload as the new state, discarding the old state.